WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -32,7 +32,37 @@ def new
   end
 
   def index
-    @competitions = Competition.all.select([:id, :name, :cityName, :countryId]).order(:year, :month, :day).reverse_order
+    query = "CAST(CONCAT(year,'-',month,'-',day) as Datetime) > ? and showAtAll = true"
+    query_params = [(Date.today - 90)]
+
+    @regions = [ ["All","all"],["",""],["Africa","_Africa"],["Asia","_Asia"],["Europe","_Europe"],["North America","_North America"],["Oceania","_Oceania"],["South America","_South America"],["",""] ] + Country.all.map { |country| [country.name, country.id] }
```

Comment:
I swear I tried, see http://pastebin.com/WtJKagTE

But it gives me an error: undefined method `name' for {:id=>"_Africa", :name=>"Africa"}:Hash